user-interface/src/staff-assignment/StaffAssignmentRow.tsx

```diff
@@ -0,0 +1,194 @@
+import { ToggleModalButton } from '@/lib/components/uswds/modal/ToggleModalButton';
+import { CaseNumber } from '@/lib/components/CaseNumber';
+import { formatDate } from '@/lib/utils/datetime';
+import { UswdsButtonStyle } from '@/lib/components/uswds/Button';
+import { useApi2 } from '@/lib/hooks/UseApi2';
+import { LoadingSpinner } from '@/lib/components/LoadingSpinner';
+import Actions from '@common/cams/actions';
+import { SearchResultsRowProps } from '@/search-results/SearchResults';
+import { AssignAttorneyModalRef, CallbackProps } from './modal/AssignAttorneyModal';
+import { useGlobalAlert } from '@/lib/hooks/UseGlobalAlert';
+import { getCaseNumber } from '@/lib/utils/formatCaseNumber';
+import { AttorneyUser } from '@common/cams/users';
+import { CaseBasics } from '@common/cams/cases';
+
+type State = {
+  assignments: AttorneyUser[];
+  isLoading: boolean;
+  bCase: CaseBasics;
+  modalRef: React.RefObject<AssignAttorneyModalRef>;
+};
+
+type Actions = {
+  getCaseAssignments: () => void;
+  updateAssignmentsCallback: (props: CallbackProps) => void;
+};
+
+export function useStaffAssignmentRowActions(initialState: State): {
+  state: State;
+  actions: Actions;
+} {
+  const api = useApi2();
+  const globalAlert = useGlobalAlert();
+
+  const [state, setState] = useState<State>(initialState);
+
+  function updateAssignmentsCallback(props: CallbackProps) {
+    const { bCase, selectedAttorneyList, previouslySelectedList, status, apiResult } = props;
+
+    if (status === 'error') {
+      globalAlert?.error((apiResult as Error).message);
+    } else if (bCase) {
+      const messageArr = [];
+      const addedAssignments = selectedAttorneyList.filter(
+        (staff) => !previouslySelectedList.find((staffObj) => staffObj.id === staff.id),
+      );
+      const removedAssignments = previouslySelectedList.filter(
+        (staff) => !selectedAttorneyList.find((staffObj) => staffObj.id === staff.id),
+      );
+
+      if (addedAssignments.length > 0) {
+        messageArr.push(
+          `${addedAssignments.map((attorney) => attorney.name).join(', ')} assigned to`,
+        );
+      }
+      if (removedAssignments.length > 0) {
+        messageArr.push(
+          `${removedAssignments.map((attorney) => attorney.name).join(', ')} unassigned from`,
+        );
+      }
+      const message =
+        messageArr.join(' case and ') + ` case ${getCaseNumber(bCase.caseId)} ${bCase.caseTitle}.`;
+
+      globalAlert?.success(message);
+      state.modalRef.current?.hide();
+      getCaseAssignments();
+    }
+  }
+
+  async function getCaseAssignments() {
+    api
+      .getCaseAssignments(state.bCase.caseId)
+      .then((response) => {
+        const assignments = response.data.map((assignment) => {
+          return { id: assignment.userId, name: assignment.name };
+        });
+        setState({ ...state, assignments, isLoading: false });
+        state.assignments = assignments;
```

Comment:
We just updated the state.  Why are we then assignment assignments to state after setting the state?

Reply:
...we can take a look at this in the experiment on StaffAssignmentRow as that code has been reorganized.